NuRadioMC/simulation/simulation.py

```diff
@@ -521,10 +537,11 @@ def run(self):
                                     100 * distance_cut_time / total_time,
                                     100 * (total_time - total_time_sum) / total_time))
 
-                    # read all quantities from hdf5 file and store them in local variables
-                    self._read_input_neutrino_properties()
+
                     logger.debug(f"simulating shower {self._shower_index}: {self._shower_type} with E = {self._shower_energy/units.eV:.2g}eV")
-                    x1 = np.array([self._x, self._y, self._z])  # the interaction point
+                    #  should be of the shower, therefore we need to update the _read_neutrino_properties to the current shower
+                    self._read_input_particle_properties()
+                    x1 = self.input_particle[simp.vertex]  # the interaction point
```

Comment:
it is not clear to me where input_particle points to? Is this now the shower? Then particle is maybe a bad name?

Reply:
right, naming is still bad, will change that, just thought the `_read_neutrino_properties` before was even worse